test/Interop/Cxx/value-witness-table/Inputs/custom-destructor.h

```diff
@@ -0,0 +1,71 @@
+struct HasUserProvidedDestructor {
+  int *value;
+  ~HasUserProvidedDestructor() { *value = 42; }
+};
+
+struct HasEmptyDestructorAndMemberWithUserDefinedConstructor {
+  HasUserProvidedDestructor member;
+  ~HasEmptyDestructorAndMemberWithUserDefinedConstructor() { /* empty */
+  }
+};
+
+struct HasNonTrivialImplicitDestructor {
+  HasUserProvidedDestructor member;
+};
+
+struct HasNonTrivialDefaultedDestructor {
+  HasUserProvidedDestructor member;
+  ~HasNonTrivialDefaultedDestructor() = default;
+};
+
+struct HasDefaultedDestructor {
+  ~HasDefaultedDestructor() = default;
+};
+
+// For the following objects with virtual bases / destructors, make sure that
+// any exectuable user of these objects disable rtti and exceptions. Otherwise,
+// the linker will error because of undefined vtables. Once we can link with
+// libc++ this requirement can be removed.
+
+struct HasVirtualBaseAndDestructor : virtual HasDefaultedDestructor {
+  int *value;
+  HasVirtualBaseAndDestructor(int *value) : value(value) {}
+  ~HasVirtualBaseAndDestructor() { *value = 42; }
+};
+
+struct HasVirtualDestructor {
+  // An object with a virtual destructor requires a deleted destructor in case
+  // we try to delete the base object. Until we can link against libc++, use
+  // this dummy implementation.
+  static void operator delete(void *p) { __builtin_unreachable(); }
+  virtual ~HasVirtualDestructor(){};
```

Comment:
```suggestion
  virtual ~HasVirtualDestructor() {}
```

Reply:
Fixed. I agree that this is better, but I also like to use `clang-format` on all my changes, and clang-format seems to disagree about this. I'm going to look into fixing our clang-format config so that this is formatted correctly.